rebalance/rebalance.py

```diff
@@ -492,7 +535,7 @@ def maybe_rebalance_pairs(plugin: Plugin, ch1, ch2, failed_channels: list):
         try:
             res = rebalance(plugin, outgoing_scid=scid1, incoming_scid=scid2,
                             msatoshi=amount, retry_for=1200, maxfeepercent=0,
-                            exemptfee=maxfee, maxhops=6, msatfactor=1)
+                            exemptfee=maxfee)
```

Comment:
Do we want `getroute_method=getroute_basic` for this? Long running rebalanceall does not need to be optimized, we want to try all routes because we have time, no?

Reply:
Twitter dup: Should we add an `getroute_lowfee` option that tries with lower fees first and then increases when no route found? That could also be awesome for rebalanceall.

Reply:
> Do we want `getroute_method=getroute_basic` for this? Long running rebalanceall does not need to be optimized, we want to try all routes because we have time, no?

I'm not sure. This way we have an option to select `getroute_method` for `rebalanceall`.

Reply:
The reason we used maxhops 6 and msatfactor 1 here before was to partly disable the 'short and big pipes first's' optimization we added per Default. I suggested this because we have plenty of time when using rebalanceall. 

Using 'first and short routes first' does make sense for manual rebalances where the user expect a successful result within 60seconds, which 'getroute_basic' hardly delivers.

I suggest we add a 'getroute_smallfee' iteration function and use that for rebalance all or go with 'getroute_basic'.

Reply:
I think it is good this way right now.
1) Most importantly: `getroute_method` is not hard-coded, it can be modified by config.
2) The default method is `iterative`: it uses big channels first. This is a good thing, not because of the time, but because most likely it will not mess up my neighbors' channels.
3) `getroute_smallfee` is a great idea. This PR makes `getroute` interchangeable, and we can implement `getroute_smallfee` later, in a different PR. Even we can make it the default `getroute_method`.

Reply:
Agreed, let me test this in production a bit and then we can merge if there are no issues...